src/sonic-frr/patch/0054-support-vrrp6-commands-and-tracking-interface.patch

```diff
@@ -0,0 +1,3091 @@
+ }
+ 
+ /*
+@@ -140,7 +143,7 @@ DEFPY_YANG(no_vrrp_priority,
+ {
+ 	nb_cli_enqueue_change(vty, "./priority", NB_OP_MODIFY, NULL);
+ 
+-	return nb_cli_apply_changes(vty, VRRP_XPATH_ENTRY, vrid);
++	return nb_cli_apply_changes(vty, VRRP_XPATH_ENTRY, vrid, VRRP_FAMILY);
+ }
+ 
+ void cli_show_priority(struct vty *vty, const struct lyd_node *dnode,
+@@ -170,7 +173,7 @@ DEFPY_YANG(vrrp_advertisement_interval,
+ 	nb_cli_enqueue_change(vty, "./advertisement-interval", NB_OP_MODIFY,
+ 			      val);
+ 
+-	return nb_cli_apply_changes(vty, VRRP_XPATH_ENTRY, vrid);
++	return nb_cli_apply_changes(vty, VRRP_XPATH_ENTRY, vrid, VRRP_FAMILY);
+ }
+ 
+ /*
+@@ -186,7 +189,7 @@ DEFPY_YANG(no_vrrp_advertisement_interval,
+ 	nb_cli_enqueue_change(vty, "./advertisement-interval", NB_OP_MODIFY,
+ 			      NULL);
+ 
+-	return nb_cli_apply_changes(vty, VRRP_XPATH_ENTRY, vrid);
++	return nb_cli_apply_changes(vty, VRRP_XPATH_ENTRY, vrid, VRRP_FAMILY);
+ }
+ 
+ void cli_show_advertisement_interval(struct vty *vty, const struct lyd_node *dnode,
+@@ -215,7 +218,7 @@ DEFPY_YANG(vrrp_ip,
+ 	int op = no ? NB_OP_DESTROY : NB_OP_CREATE;
+ 	nb_cli_enqueue_change(vty, "./v4/virtual-address", op, ip_str);
+ 
+-	return nb_cli_apply_changes(vty, VRRP_XPATH_ENTRY, vrid);
++	return nb_cli_apply_changes(vty, VRRP_XPATH_ENTRY, vrid, VRRP_FAMILY);
+ }
+ 
+ void cli_show_ip(struct vty *vty, const struct lyd_node *dnode, bool show_defaults)
+@@ -229,30 +232,62 @@ void cli_show_ip(struct vty *vty, const struct lyd_node *dnode, bool show_defaul
+ 
+ /*
+  * XPath:
+- * /frr-interface:lib/interface/frr-vrrpd:vrrp/vrrp-group/v6/virtual-address
++ * /frr-interface:lib/interface/frr-vrrpd:vrrp/vrrp-group/track-interface
+  */
+-DEFPY_YANG(vrrp_ip6,
+-      vrrp_ip6_cmd,
+-      "[no] vrrp (1-255)$vrid ipv6 X:X::X:X",
++DEFPY_YANG(vrrp_track_interface,
++      vrrp_track_interface_cmd,
++      "vrrp (1-255)$vrid track-interface IFNAME$intf_name priority-dec (10-50)",
++      VRRP_STR
++      VRRP_VRID_STR
++      "Add tracking interface\n"
++      VRRP_TRACK_INTF_STR
++      VRRP_PRIORITY_DECREMENT_STR
++      "Priority decrement value\n")
++{
++	char xpath[XPATH_MAXLEN];
++	char xpath_track[XPATH_MAXLEN + 32];
++
++	snprintf(xpath, sizeof(xpath),
++			"./track-interface[interface='%s']",
++			intf_name);
++	nb_cli_enqueue_change(vty, xpath, NB_OP_CREATE, NULL);
++
++	snprintf(xpath_track, sizeof(xpath_track),
++			"%s/priority-decrement", xpath);
++	nb_cli_enqueue_change(vty, xpath_track, NB_OP_MODIFY, priority_dec_str);
++
++	return nb_cli_apply_changes(vty, VRRP_XPATH_ENTRY, vrid, VRRP_FAMILY);
++}
++
++DEFPY_YANG(no_vrrp_track_interface,
++      no_vrrp_track_interface_cmd,
++      "no vrrp (1-255)$vrid track-interface IFNAME$intf_name",
+       NO_STR
+       VRRP_STR
+       VRRP_VRID_STR
+-      "Add IPv6 address\n"
+-      VRRP_IP_STR)
++      "Add tracking interface\n"
++      VRRP_TRACK_INTF_STR
++	  VRRP_PRIORITY_DECREMENT_STR
```

Comment:
Do we need VRRP_PRIORITY_DECREMENT_STR here?

Reply:
It will be removed, thanks.